lua/controller.lua

```diff
@@ -334,7 +342,12 @@ local function controller_allow_metadata_inventory_put(pos, listname, index, sta
 		local drawer = drawer_net_index[stack:get_name()]
 
 		if drawers.drawer_get_content(drawer.drawer_pos, drawer.visualid).name == stack:get_name() then
-			return drawers.drawer_can_insert_stack(drawer.drawer_pos, stack, drawer["visualid"])
+			local can_store = drawers.drawer_can_insert_stack(drawer.drawer_pos, stack, drawer["visualid"])
+			-- If we have a drawer for this item, but can't put anything there, it's probably full
+			-- attempt to put in a new empty drawer
+			if can_store ~= 0 then
+				return can_store
+			end
```

Comment:
The important part here is the implicit else, which will allow the instruction pointer to flow into the following section to see if it can fit the stack into an empty drawer.